pkg/ovs/ovsconfig/ovs_client.go

```diff
@@ -783,12 +787,29 @@ func (br *OVSBridge) GetOVSVersion() (string, Error) {
 		klog.Error("Transaction failed: ", err)
 		return "", NewTransactionError(err, temporary)
 	}
+
 	if len(res[0].Rows) == 0 {
-		klog.Warning("Could not find ovs_version")
-		return "", nil
+		klog.Warning("Could not find ovs_version in the OVS query result")
+		return "", NewTransactionError(err, false)
+	} else {
+		return ParseOvsVersion(res[0].Rows)
 	}
+}
 
-	return res[0].Rows[0].(map[string]interface{})["ovs_version"].(string), nil
+func ParseOvsVersion(ii interface{}) (string, Error) {
+	errorMessage := fmt.Errorf("unexpected return from Ovs.  You might try running '%v' inside a container to correct this error", GetOvsVersionMessage())
+	switch ii.(type) {
+	default:
+	case map[string]string:
+		if _, ok := ii.(map[string]string)["ovs_version"]; ok {
+			return ii.(map[string]string)["ovs_version"], nil
+		}
+	case map[string]interface{}:
+		if _, ok := ii.(map[string]interface{})["ovs_version"]; ok {
+			return ii.(map[string]interface{})["ovs_version"].(string), nil
+		}
+	}
```

Comment:
what do the 2 different cases correspond to?

Reply:
seems like either is a valid way to return a map w the ovs version in it , and there doesnt seem to be any gaurantee on the type returned here .